zerver/lib/request.py

```diff
@@ -367,6 +369,18 @@ def _wrapped_view_func(request: HttpRequest, *args: Any, **kwargs: Any) -> HttpR
 
             kwargs[func_var_name] = val
 
+        request._unsupported_params = []
+
+        if hasattr(request, 'method'):
+            req_params = set()
+            if request.method == 'POST':
+                req_params = set(request.POST.keys())
+            elif request.method == 'GET':
+                req_params = set(request.GET.keys())
+
+            if len(req_params) > 0:
+                request._unsupported_params = list(req_params - used_params)
```

Comment:
Actually, having thought about this a bit more, we need to make sure this logic supports the case where we call `has_request_variables` multiple times on the same HttpRequest.  I think the right way to do that is this:

* `request._req_user_parameters` should be initialized as an empty set only if not already present, above.
* We add things as we go.
* The logic to show these extra parameters lives in `json_success` / `json_response`.  I would have liked to do this via middleware, but those functions are what converts us to use an HttpResponse.

Let me start a thread on this issue in #backend.